product_variant_default_code/models/product.py

```diff
@@ -123,22 +123,26 @@ def create(self, vals):
         return super(ProductTemplate, self).create(vals)
 
     def write(self, vals):
+        mask_templates = self.env['product.template']
         product_obj = self.env['product.product']
         if ('reference_mask' in vals and not vals['reference_mask'] or not
                 self.user_has_groups(
                     'product_variant_default_code.group_product_default_code'
                 )):
-            if self.attribute_line_ids:
-                vals['reference_mask'] = self._get_default_mask()
-        result = super(ProductTemplate, self).write(vals)
+            for template in self.filtered("attribute_line_ids"):
```

Comment:
No need of keeping another recordset

```suggestion
            with_variants = self.filtered("attribute_line_ids")
            for template in variants:
                ...
        super(ProductTemplate, self - with_variants).write(vals)
```

Reply:
Right, thanks @pedrobaeza , changes done